public/src/components/epicTests/newNameCreator.tsx

```diff
@@ -38,40 +38,52 @@ class NewNameCreator extends React.Component<NewNameCreatorProps, NewNameCreator
     errorMessage: ""
   }
 
-  onNewNameButtonClick = (event: React.MouseEvent<HTMLButtonElement>) =>  {
+  onNewNameButtonClick = (event: React.MouseEvent<HTMLButtonElement>): void =>  {
     this.setState({
       newNamePopoverOpen: true,
       anchorElForPopover: event.currentTarget
     });
   }
 
-  handleCancel = () => {
-    this.setState({ newNamePopoverOpen: false });
+  closePopover = (): void => {
+    this.setState(
+      {
+        newNamePopoverOpen: false,
+        errorMessage: ""
+      }
+    );
   }
 
-  isDuplicateName = (newName: string) => {
+  setErrorMessage = (message: string): void => {
+    this.setState( {errorMessage: message });
+  };
+
+  isDuplicateName = (newName: string): boolean => {
     const newLowerCase: string = newName.toLowerCase();
     const isDuplicate = this.props.existingNames.some(name => name.toLowerCase() === newLowerCase);
     return isDuplicate;
-  }
+  };
 
-  handleName = (newTestName: string) => {
+  containsInvalidChars = (newName: string): boolean => {
+    return (/[^\w-]/.test(newName));
```

Comment:
we should allow `_` as well

Reply:
sorry, just realised `\w` covers this! Perfect!

Reply:
Yep, that's including in `\w`: Matches any alphanumeric character including the underscore. Equivalent to [A-Za-z0-9_].